### PR TITLE
CLOUDSTACK-10070: Fixing some component tests and adding them in travis

### PR DESCRIPTION
**Pull request by borisstoyanov**
ping @rhtyd @DaanHoogland @nvazquez @PaulAngus for review

### CODE REVIEW

**Comment by borisstoyanov** [comment]
@blueorangutan package

File: .travis.yml [hunk 1]
```diff
@@ -100,15 +100,19 @@ env:
              component/test_acl_listvm
              component/test_acl_listvolume
              component/test_acl_sharednetwork
-             component/test_acl_sharednetwork_deployVM-impersonation"
+             component/test_acl_sharednetwork_deployVM-impersonationa
```
Comment (rhtyd) [inline .travis.yml]: @borisstoyanov is there a typo at the end of the test/file name `a`?

**Comment by rhtyd** [comment]
@borisstoyanov there seems to be some travis job failures, please see

**Comment by borisstoyanov** [comment]
I got this in the logs: 
```
The job exceeded the maximum time limit for jobs, and has been terminated.
```
Looks like I'll need to rearrange the execution batch definitions in the yml. Can I just add them in new jobs?

File: .travis.yml (hunk 1 above)
Comment (borisstoyanov) [reply .travis.yml]: yes will fix

**Comment by borisstoyanov** [comment]
@rhtyd current travis failure is not related to the changes I've made

**Comment by rhtyd** [comment]
@borisstoyanov okay I've rekicked the failing job, can you compare the test time take for each of the jobs and try to consolidate them (rearrange tests) such that each job may be completed in less or equal to 50 minutes? Some tests that run in 20-25 minutes are prime targets. The first job may be refactored to get more tests as well (that run quite quickly?).

**Comment by rhtyd** [comment]
@borisstoyanov I've rekicked one of the failing travis jobs, in general LGTM.
@blueorangutan package

**Comment by borisstoyanov** [comment]
@rhtyd the changes are in /component/.. do you think it's required to run a smoketest?

**Comment by rhtyd** [comment]
@borisstoyanov good point, can you run a component test run to confirm regression testings?

**Comment by borisstoyanov** [comment]
@rhtyd those tests pass in travis do you want explicit run on KVM env?

**Comment by rhtyd** [comment]
@borisstoyanov yes, since we're changing test code, let's validate them against one of the hypervisors; kvm would be a good candidate.

**Comment by borisstoyanov** [comment]
@blueorangutan help

**Comment by borisstoyanov** [comment]
@blueorangutan test centos7 kvm-centos7 component/test_affinity_groups_projects.py, component/test_cpu_domain_limits.py, component/test_organization_states.py, component/test_project_resources.py, component/test_vpc.py

**Comment by borisstoyanov** [comment]
ping @rhtyd results looks good to me

**Comment by rhtyd** [comment]
@borisstoyanov we'll have to re-run the tests, I fixed the issue the additional tests need to be space separated, just fixed BO's help/doc
@blueorangutan test centos7 kvm-centos7 component/test_affinity_groups_projects.py component/test_cpu_domain_limits.py component/test_organization_states.py component/test_project_resources.py component/test_vpc.py

**Comment by rhtyd** [comment]
@borisstoyanov please see/fix failing component test failures

**Comment by borisstoyanov** [comment]
test_organizational_states is failing due to this big: https://issues.apache.org/jira/browse/CLOUDSTACK-7735

the failures in test_cpu_domain_limits.py are due to not merged changes in Trillian: https://github.com/shapeblue/Trillian/pull/39

**Comment by rhtyd** [comment]
@borisstoyanov okay, I've merged the requested Trillian PR, can you re-run the test ?

**Comment by borisstoyanov** [comment]
thanks @rhtyd 
@blueorangutan test centos7 kvm-centos7 component/test_affinity_groups_projects.py component/test_cpu_domain_limits.py component/test_organization_states.py component/test_project_resources.py component/test_vpc.py

**Comment by rhtyd** [comment]
@borisstoyanov can you fix the conflicts?

**Comment by borisstoyanov** [comment]
Done @rhtyd

**Comment by rhtyd** [comment]
LGTM @borisstoyanov, I've remove duplicates from .travis.yml and re-sorted them. Can you re-run tests on changed components tests one more time?

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
thanks @rhtyd
@blueorangutan test centos7 kvm-centos7 component/test_affinity_groups_projects.py component/test_cpu_domain_limits.py component/test_organization_states.py component/test_project_resources.py component/test_vpc.py

**Comment by borisstoyanov** [comment]
This doesn't look right, will re-run them...
@blueorangutan test centos7 kvm-centos7 component/test_affinity_groups_projects.py component/test_cpu_domain_limits.py component/test_organization_states.py component/test_project_resources.py component/test_vpc.p

**Comment by rhtyd** [comment]
@borisstoyanov the additional tests are still failing, are you planning to work on them or remove the test_organization_states.py from this PR?

**Comment by borisstoyanov** [comment]
@rhtyd it's the allocation_state.py that it's failing, let have a quick look. As far as organization_state.py, this is already dependant to a bugfix commented in the .travis yaml

File: test/integration/component/test_organization_states.py [hunk 2]
```diff
@@ -501,17 +501,17 @@ def test_41_disableHost(self):
 	Disable Host
         Validate that listHosts() returns the  allocationstate as "Disabled"
 	"""
-	self.debug("Host to be disabled: " + self.host.id)
+        self.debug("Host to be disabled: " + self.host.id)
```
Comment (rhtyd) [inline test/integration/component/test_organization_states.py]: Looks like this adds an additional ` ` whitespace, and changes below.

**Comment by rhtyd** [comment]
@borisstoyanov fair enough, can you look at one of the Travis job failures?

**Comment by borisstoyanov** [comment]
yes, allocation_state is failing, looking at it now..

**Comment by rhtyd** [comment]
okay @borisstoyanov I'll rearrange and try to fix the tests, when next Travis run passes I'll merge this.

**Comment by rhtyd** [comment]
Tested and applied a fix manually:
==== Marvin Init Successful ====
=== TestName: test_01_zones | Status : SUCCESS ===

=== TestName: test_02_pods | Status : SUCCESS ===

=== TestName: test_03_clusters | Status : SUCCESS ===

=== TestName: test_04_hosts | Status : SUCCESS ===

=== TestName: test_05_storage_pools | Status : SUCCESS ===

===final results are now copied to: /tmp//MarvinLogs/test_allocation_states_42K6QY===

Merging now.

**Comment by rhtyd** [comment]
Travis is green, merging now.

**Pull request merged**